src/symbol_table.rs

```diff
@@ -69,6 +81,18 @@ impl SymbolTable {
         sid
     }
 
+    /// If `maybe_text` is `Some(text)`, this method is equivalent to `add_symbol(text)`.
+    /// If `maybe_text` is `None`, this method is equivalent to `add_placeholder()`.
+    pub(crate) fn add_symbol_or_placeholder<A: AsRef<str>>(
+        &mut self,
+        maybe_text: Option<A>,
+    ) -> SymbolId {
+        match maybe_text {
+            Some(text) => self.add_symbol(text),
+            None => self.add_placeholder(),
+        }
+    }
+
     /// If `maybe_text` is `Some(text)`, this method is equivalent to `intern(text)`.
     /// If `maybe_text` is `None`, this method is equivalent to `add_placeholder()`.
     pub(crate) fn intern_or_add_placeholder<A: AsRef<str>>(
```

Comment:
Is this method still being used anywhere?

Reply:
Yes its being used in `SymbolTable#initalize`.